components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/TokenValidationHandler.java

```diff
@@ -440,7 +441,15 @@ private OAuth2IntrospectionResponseDTO validateAccessToken(OAuth2TokenValidation
 
         } else {
             try {
-                accessTokenDO = OAuth2Util.findAccessToken(validationRequest.getAccessToken().getIdentifier(), false);
+                int tenantId = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantId();
+                accessTokenDO = OAuth2Util.findAccessToken(validationRequest.getAccessToken().getIdentifier(),
```

Comment:
At this point shouldn't this DO object be available in the context?
Can we see if we could avoid this lookup?

Reply:
No we don't have the DO object available.